feat(addons): update addons documentation (#29)
anbox-cloud/discourse-docs: addons-tutorial.md

Addons are used to customize the images used for the containers. An addon has *hooks* that are
invoked at various points in the lifecycle of a container. A good example is an addon which
enables the SSH service inside a container to provide access from the outside for any kind
of automation of the container.  

> Addons are created independently, and can be attached to individual applications.

Reviewer: I think this will work nicely as a tutorial, great idea! :)
We don't have any "real" tutorials in the docs yet, so there isn't much to refer to. But overall, we don't want a lot of explanation in here - we just want to guide users through the steps to get a result.

So the introduction should be a bit more direct. Not explaining why addons are nice and what you can do with them, but telling users that in this tutorial, they will create an addon with a hook that enables the SSH service.

> <h2 id='heading--build-your-own-addon'>Build your first addon</h2>

Reviewer: This heading isn't needed imo. That is what the whole tutorial is about. :wink: 

I suggest to use numbered headings here, one for every step that the user should do. So:
1. Create a manifest file
2. Add a pre-start hook
3. and 4. as is

Any addon must have a `manifest.yaml` file and at least one hook in the `hooks` directory, e.g:

```text
manifest.yaml
hooks/pre-start

> ```

Reviewer: Not needed here - we have it in the reference already, and the user will see it more clearly after creating the files.

### Metadata
The `manifest.yaml` file contains metadata information about your addon.
Create a  `manifest.yaml` file that contains the following

```yaml
name: ssh
description: |  
	Enable SSH access when starting a container  
```

### Hooks
The `hooks` directory contains executable files where each corresponds to a single hook. Hooks can be implemented in any language, but we are using a bash script here.

Create a directory named `hooks` and, in it, a file named `pre-start` containing the following script:
```bash
#!/bin/bash

systemctl start sshd
```

> INFO: supported hooks are: `pre-start`, `post-start` and `post-stop`. Read more about them [here](TODO: INSERT LINK TO ADDON REFERENCE)

### Create the addon

> Assuming your addon structure currently looks like this:

Reviewer: No need to assume - this is what we have created. :wink: 
It makes sense to show the structure here imo, but introduce with something like "This is the addon structure you  created". You could also add a link to the file structure section in the reference here.

```bash
enable-ssh
├── hooks
│   └── pre-start
└── manifest.yaml
```  

run the following command (this may take a minute):
```bash
amc addon add ssh ./enable-ssh
```

When your addon is created, you can view it with
```bash
amc addon list
```

### Use the addon in an application

Create an application manifest to include the addon name under `addons`

```yaml
name: my-application
instance-type: a2.3
addons:
  - ssh
```

Then create your application
```bash
application_id=$(amc application create ./my-application)
amc wait "$application_id" -c status=ready
```

The `amc wait` command returns when your application is ready to launch.  
You can now launch an instance of your application.
```bash
amc launch -s +ssh my-application
```
> NOTE: port 22 (ssh) is closed by default. We open it via services. Read more about services [here](https://anbox-cloud.io/docs/howto/container/expose-services)

> You should now be able to access your container via ssh.

Reviewer: ```suggestion
You can now access your container via ssh.
```
We shouldn't expect anything to go wrong in a tutorial. ;)
